### DISCUSSION TITLE
Using Stockfish evaluation function as library

### DISCUSSION THREAD

**Discussion by delphi-coder** [official-stockfish/Stockfish#4054] [Q&A]
Hi,
I need trust-able evaluation function to use in engine development for debugging purpose. What I'm trying to do is passing 12 bitboards + castling righs + fifty move variable + ep pos + side to move into DLL function and use static scores in my own search algorithm. I'm not familiar with stockfish source but if i understand correctly I need to setup Position class and pass it to evaluate and then return the Integer score for it.
Now the question is:
1- How to pass 64 bit pieces bitboards (White Rooks,White Bishops,....,Black Rooks,...) to setup Position class? (not by using FEN)
2- returns value of evaluate is enum, its not simply an integer number as score, How to turn it to integer score?
3- I picked up Stockfish 2.3 source as I think it should be easier to read and compile under VS2010. Any idea about this would be appreciated.

**Comment by atumanian** [comment]
Hello, Delphi-Coder!
I'll try to answer some of your questions. There are `Position::put_piece` and `Position::remove_piece` methods. But these are not sufficient to set up a position. You also need `set_castling_right`, `set_state` and `set_check_info` methods of the `Position` class, but unfortunately they are private. By the way, you can use the `eval` command-line interface command.
You can simply cast a Value to an integer: `int(v)`
